spacewalk/certs-tools/rhn_bootstrap_strings.py

```diff
@@ -406,6 +406,11 @@ def getRegistrationStackSh(saltEnabled):
     CLIENT_REPO_NAME="susemanager:bootstrap"
     CLIENT_REPO_FILE="/etc/yum.repos.d/$CLIENT_REPO_NAME.repo"
 
+    # In case of CentOS, check is centos bootstrap repository is available, if not, failback to res.
+    if [[ "$Y_CLIENT_CODE_BASE" == centos && ! `$FETCH $CLIENT_REPO_URL/repodata/repomd.xml` ]]; then
```

Comment:
Does `$FETCH` print any value on screen?
Not sure if you need `&> /dev/null`

Reply:
The `$FETCH` is wrapping `curl` or `wget` with silent options: i.a. `curl -ksSOf`.

The only output we would see, in case the "centos" repo is not available would be something like this:

```
https://my-suma-server/pub/repositories/centos/6/bootstrap/repodata/repomd.xml:
2020-07-01 13:27:35 ERROR 404: Not Found.
```
Then you will see yum trying to use the "res" repository.

Reply:
Do we really want to show it if anyway the RES repo is going to be used?

I think users can be alarmed if we see that.

Reply:
If I may suggest a change:
```
if [[ "$Y_CLIENT_CODE_BASE" == centos && ! `$FETCH $CLIENT_REPO_URL/repodata/repomd.xml` ]]; then
    echo "CentOS${{Y_CLIENT_CODE_VERSION}} bootstrap repository not found, using RES${{Y_CLIENT_CODE_VERSION}} bootstrap repository instead"
    CLIENT_REPO_URL="${{CLIENT_REPOS_ROOT}}/res/${{Y_CLIENT_CODE_VERSION}}/bootstrap"
fi

Reply:
> Do we really want to show it if anyway the RES repo is going to be used?
> 
> I think users can be alarmed if we they that.

The only possible output from `curl`, when the "centos" repo does not exist, would be only showing "centos" and never mentining "res" (since `curl` always and only fetchs the "centos" repo). 

So piping "curl" output to `/dev/null` here is not really helpful. Actually, I think it makes sense to keep the possible output so the users knows the bootstrap script initially tries to use the "centos" repos and then it goes with the "res" one, and not doing the magic completely silently.

Reply:
Ok. I didn't notice the `echo` you suggested! You're right. Changes are now pushed :+1: